addons/l10n_tr_nilvera_edespatch/models/res_partner.py

```diff
@@ -0,0 +1,31 @@
+from odoo import _, models
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    def _l10n_tr_nilvera_validate_partner_details(self):
+        error_messages = {}
+        for record in self:
+            required_fields = {
+                "street": record.street,
+                "city": record.city,
+                "state": record.state_id,
+                "zip": record.zip,
+                "country": record.country_id,
+            }
+            missing_fields = [field.capitalize() for field, value in required_fields.items() if not value]
```

Comment:
```suggestion
            missing_fields = [field_name for field_name, value in required_fields.items() if not value]
```
using capitalize might mess with translator better to keep it as translate when assigning it as key